python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -335,6 +350,75 @@ def toposort_asset_keys(self) -> Sequence[AbstractSet[AssetKey]]:
     def has_self_dependency(self, asset_key: AssetKey) -> bool:
         return asset_key in self.get_parents(asset_key)
 
+    def bfs_filter_subsets(
+        self,
+        condition_fn: Callable[[AssetKey, Optional[PartitionsSubset]], bool],
+        initial_subset: "AssetGraphSubset",
+    ) -> "AssetGraphSubset":
+        """
+        Returns asset partitions within the graph that
+        - Are >= initial_asset_partitions
+        - Asset matches the condition_fn
+        - Any of their ancestors >= initial_asset_partitions match the condition_fn
+
+        Visits parents before children.
+        """
+        from .asset_graph_subset import AssetGraphSubset
+
+        all_assets = set(initial_subset.asset_keys)
+        check.invariant(
+            len(initial_subset.asset_keys) == 1, "Multiple initial assets not yet supported"
```

Comment:
This might be unrelated but this seems like it still expects only a single root asset 👀